lib/utils/middlewares/client_ip_middleware.ts

```diff
@@ -7,13 +7,20 @@ import { IRequestMiddleware } from "../request_middleware";
 
 export type ClientIp = Option<IPString>;
 
+import * as winston from "winston";
+
 /**
- * A middleware that extracts the client IP from the request
+ * A middleware that extracts the client IP from the request.
+ *
+ * The IP is extracted from the "x-forwarded-for" HTTP header;
```

Comment:
That's not correct, [here's how it works](https://www.npmjs.com/package/request-ip#how-it-works) - if you want you can just link to that page

Reply:
ok, but that's how it works in our context atm (using azure api management). 
the point of this PR is that I got a generic "403 forbidden" once (because I've called the functions code directly bypassing the api gateway) and took it 1hr to debug why :)

Reply:
yes this is how the system uses this function, but it's not how this function works :)

so perhaps you should link to the `request-ip` documentation from here, and add the comment about `x-forwarded-for` to the README

Reply:
done